server/services/store/mattermostauthlayer/mattermostauthlayer.go

```diff
@@ -500,3 +503,81 @@ func (s *MattermostAuthLayer) CreatePrivateWorkspace(userID string) (string, err
 
 	return channel.Id, nil
 }
+
+func (s *MattermostAuthLayer) GetFileInfo(id string) (*mmmodel.FileInfo, error) {
+	fileInfo, appErr := s.pluginAPI.GetFileInfo(id)
+	if appErr != nil {
+		// Not finding fileinfo is fine because we don't have data for
+		// any existing files already uploaded in Boards before this code
+		// was deployed.
+		if appErr.StatusCode == http.StatusNotFound {
+			return nil, nil
+		}
+
+		s.logger.Error("error fetching fileinfo", mlog.String("id", id), mlog.Err(appErr))
+		return nil, appErr
+	}
+
+	return fileInfo, nil
+}
+
+func (s *MattermostAuthLayer) SaveFileInfo(fileInfo *mmmodel.FileInfo) error {
```

Comment:
Why is `GetFileInfo` a pluginAPI, but `SaveFileInfo` we do directly. I would rather see both as plugin api calls.

Reply:
I'm working on the usages API - https://docs.google.com/document/d/19yqSXo-8dahwITBYrkrP9Qy1S28J0o9u8ZBrmCugJv8/edit#

I need to track the storage usage for cloud freemium. 

Doing a `select sum(size) from FileInfo` on the server every time the API is called could end up being a problem for a large `FileInfo` table. 

I was going to store this value in the systems table and update it on every file upload, but if plugins directly write to the FileInfo table, then the files uploaded by the plugins wont be counted in the storage usage. 
If we have a `SaveFileInfo` plugin API endpoint, then I can correctly track storage usage.

For now I can create a cached version that does a `select sum(size) from FileInfo` every hour. I don't think we need real-time file usage tracking just yet.

Reply:
@ashishbhate we've discussed this in DM. LMK if there is anything else I can do.